layers/+lang/extempore/packages.el

```diff
@@ -0,0 +1,97 @@
+;;; packages.el --- extempore layer packages file for Spacemacs.
+;;
+;; Copyright (c) 2020 Ben Swift
+;;
+;; Author: Ben Swift <ben@benswift.me>
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+
+;;; Commentary:
+
+;; See the Spacemacs documentation and FAQs for instructions on how to implement
+;; a new layer:
+;;
+;;   SPC h SPC layers RET
+;;
+;;
+;; Briefly, each package to be installed or configured by this layer should be
+;; added to `extempore-packages'. Then, for each package PACKAGE:
+;;
+;; - If PACKAGE is not referenced by any other Spacemacs layer, define a
+;;   function `extempore/init-PACKAGE' to load and initialize the package.
+
+;; - Otherwise, PACKAGE is already referenced by another Spacemacs layer, so
+;;   define the functions `extempore/pre-init-PACKAGE' and/or
+;;   `extempore/post-init-PACKAGE' to customize the package as it is loaded.
+
+;;; Code:
+
+(defconst extempore-packages
+  '(extempore-mode)
+  "The list of Lisp packages required by the extempore layer.
+
+Each entry is either:
+
+1. A symbol, which is interpreted as a package to be installed, or
+
+2. A list of the form (PACKAGE KEYS...), where PACKAGE is the
+    name of the package to be installed or loaded, and KEYS are
+    any number of keyword-value-pairs.
+
+    The following keys are accepted:
+
+    - :excluded (t or nil): Prevent the package from being loaded
+      if value is non-nil
+
+    - :location: Specify a custom installation location.
+      The following values are legal:
+
+      - The symbol `elpa' (default) means PACKAGE will be
+        installed using the Emacs package manager.
+
+      - The symbol `local' directs Spacemacs to load the file at
+        `./local/PACKAGE/PACKAGE.el'
+
+      - A list beginning with the symbol `recipe' is a melpa
+        recipe.  See: https://github.com/milkypostman/melpa#recipe-format")
+
+(defun extempore/init-extempore-mode ()
+  "Initialize extempore mode"
+  (use-package extempore-mode
```

Comment:
You should defer loading of `extempore-mode` by adding `:defer t` here.
If `extempore-mode` does already autoloading all should still be working. If not you need to declare proper autoloads with use-package.